pkg/controllers/bgp/bgpconf_controller.go

```diff
@@ -190,9 +246,28 @@ func (r *BgpConfReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	// The policy field must be indexed by the manager, so that we will be able to lookup BgpConf by a referenced ConfigMap name.
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1alpha2.BgpConf{}, policyField, func(rawObj runtime.Object) []string {
+		// Extract the ConfigMap name from the BgpConf Spec, if one is provided
+		bgpConf := rawObj.(*v1alpha2.BgpConf)
+		if bgpConf.Spec.Policy == "" {
+			return nil
+		}
+		return []string{bgpConf.Spec.Policy}
+	})
+	if err != nil {
+		return err
+	}
+
 	ctl, err := ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha2.BgpConf{}).
 		WithEventFilter(p).
+		Owns(&appsv1.Deployment{}).
```

Comment:
Just a question, not an objection, why use `Owns(&appsv1.Deployment{})`?

Reply:
Since we create dependency Deployments during the reconciliation, we can specify that the controller `Owns` Deployments.

Reply:
> Since we create dependency Deployments during the reconciliation

During bgpconf reconciliation.
I don't see a place to create a `Deployment`

Reply:
I think if you want to use `Owns`, then bgpconf must be the owner of `deployment`

https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#setup

Reply:
Thanks for the link :smile:. Originally, I had planned to make the Reconcile loop of the policy ConfigMap independent in some way or another of the BgpConf reconciliation. For now, I have made it a part of the BgpConf reconciliation loop. This removes the need for ownership since I will be using the existing one.